src/core/xfa/template.js

```diff
@@ -2257,7 +2257,7 @@ class Image extends StringObject {
     };
 
     if (this.href) {
-      html.attributes.src = new URL(this.href);
+      html.attributes.src = new URL(this.href).href;
```

Comment:
I think that for now we shouldn't create an image in case of `this.href !== null`:
  - the ref can be a path (a name) to an internal picture which is not in the xml stream and we'll handle this case
  - it can be a real `http://` link or whatever. In this case, what's the policy ? especially if the url has the file protocol.
  - it can be garbage as you saw it in the mentioned pdf.
And I've the same question for `a` tag in `xhtml.js`.

Reply:
The *main* objective of the PR was simply to avoid this line completely breaking everything, because of structured cloning errors (since a `URL` can't be cloned), and further improvements can probably happen elsewhere?

> * it can be a real `http://` link or whatever. In this case, what's the policy ? especially if the url has the file protocol.

I'm not sure that we should even attempt to support that case at all, given the possibly security implications of loading arbitrary content; this feels a bit similar to e.g. https://github.com/mozilla/pdf.js/issues/13266#issuecomment-824046524.